src/main/java/net/backupcup/mcde/mixin/AnvilScreenHandlerMixin.java

```diff
@@ -66,7 +66,7 @@ public AnvilScreenHandlerMixin(ScreenHandlerType<?> type, int syncId, PlayerInve
                     entry.setValue(entry.getValue() + 1);
                 }
             }
-            if (!map.isEmpty()) {
+            if (!map.isEmpty() || levelCost.get() > 0) {
```

Comment:
Never trust Java to compile out correctly when it comes to `II`, `&&`, etc. consider:
```java
if (!map.isEmpty() || (levelCost.get() > 0)) {...
```